doc/source/whatsnew/v0.21.0.txt

```diff
@@ -130,6 +130,8 @@ Numeric
 Categorical
 ^^^^^^^^^^^
 
+- Bug in ``Categorical.is_dtype_equal()`` where comparison with Series whose dtype is 'category' is not handled correctly (:issue:`16659`)
```

Comment:
comparison with a ``Series`` with ``dtype='category'``. 

remove the 'is not handled correctly' (that's what the Bug indicates.